connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -167,4 +169,26 @@ message Expression {
     // (Optional) Alias metadata expressed as a JSON map.
     optional string metadata = 3;
   }
+
+  message LambdaFunction {
+    // (Required) The lambda function.
+    Expression function = 1;
+
+    // (Required) Function arguments. Must not be empty.
+    //
+    // All arguments must be UnresolvedNamedLambdaVariables.
+    repeated Expression arguments = 2;
```

Comment:
If all arguments must be UnresolvedNamedLambdaVariables. why not define this as 
`repeated UnresolvedNamedLambdaVariables arguments = 2`?

Reply:
I was hitting something weird if using `UnresolvedNamedLambdaVariables ` here

Reply:
In the scala side, a LambdaFunction actually accepts `NamedExpression`s as argument;
In PySpark, an argument is always a `UnresolvedNamedLambdaVariable`;

Here we use `Expression` instead of `UnresolvedAttributes`, a benefit I can image is, if we need to support more types of Expression, we do not need to change the proto

Reply:
Shall we simply use `repeated String ...`? If we think about the SQL API `exists(array_col, c -> ...)`, what we need to provide here is the argument name list, which is really just a list of string.

Reply:
yes, that's simpler. At least `repeated String` is enough for existing pyspark's implementations. will update